debezium-core/src/main/java/io/debezium/jdbc/JdbcConnection.java

```diff
@@ -97,7 +97,7 @@
      * @param variables any custom or overridden configuration variables
      * @return the connection factory
      */
-    public static ConnectionFactory patternBasedFactory(String urlPattern, Field... variables) {
+    public static ConnectionFactory patternBasedFactory(String urlPattern, ClassLoader classloader, Field... variables) {
```

Comment:
Correct the JavaDoc, and specify that `classloader` may be null (see below). Also, we need to keep the old version of this method for backward compatibility, but it can simply delegate to this new signature.